### PR TITLE
Update documentation of p::s::Triangulation.

### PR DESCRIPTION
**Pull request by bangerth**
While there, also add a test to educate myself about how all of this works.

### CODE REVIEW

**Comment by drwells** [comment]
Relates #4704.

File: include/deal.II/distributed/shared_tria.h [hunk 1]
```diff
@@ -112,12 +144,42 @@ namespace parallel
...
+         *         cell->set_subdomain_id(assignment[index%16]);
+         *     }
+         *
+         *     int main ()
+         *     {
+         *       parallel::shared::Triangulation<dim> tria(...,
+         *                                                 parallel::shared::Triangulation<dim>::Settings::partition_custom_signal);
+         *       tria.signals.post_refinement.connect ([&tria]()
+         *                                             {
+         *                                               // user defined partitioning scheme
```
Comment (tjhei) [inline include/deal.II/distributed/shared_tria.h]: I don't understand this code example. Wouldn't you want to move the code from ``mypartition`` inside the lambda?

File: include/deal.II/distributed/shared_tria.h [hunk 2]
```diff
@@ -112,12 +144,42 @@ namespace parallel
          *     {
          *       parallel::shared::Triangulation<dim> tria(...,
          *                                                 parallel::shared::Triangulation<dim>::Settings::partition_custom_signal);
-         *       tria.signals.post_refinement.connect (std::bind(&mypartition<dim>, std::ref(shared_tria)));
+         *       tria.signals.post_refinement.connect (std::bind(&mypartition<dim>, std::ref(tria)));
+         *     }
+         *  @endcode
+         *
+         * An equivalent code using C++11 lambda functions would look like this:
```
Comment (tjhei) [inline include/deal.II/distributed/shared_tria.h]: "using a lambda function"? I don't think it is important to mention c++11.

**Comment by bangerth** [comment]
Yes, I expanded the code example in the wrong place. Take another look now.

**Pull request merged**